### PR TITLE
Cortx 30428 retry connection timeout support for s3bench

### PR DESCRIPTION
**Pull request by DhobaleVishal**
## Problem Statement

-   Problem statement

## Design

-   For Bug, Describe the fix here.
-   For new test addition post the link for design

## Coding

   Checklist for Author

-   [x]  Coding conventions are followed and code is formatted (e.g. using pycharm in-built formatter)

## Testing

  Checklist for Author

-   [x]  New/Affected tests are executed
-   [ ]  Attach test execution logs

## Review 

  Checklist for Author

-   [x]  JIRA number/GitHub Issue added to PR
-   [x]  PR is self reviewed
-   [x]  Jira and state/status is updated and JIRA is updated with PR link
-   [x]  Check if the description is clear and explained

## Documentation

  Checklist for Author

-   [ ]  Changes done to ReadMe

### CODE REVIEW

**Review by VikasKumarC — APPROVED** [review]
Good to merge as per requirements.
Option is on user to set given parameters otherwise default will be used.

File: workload/s3/mix_io/type4_mix_object_crud_operations.yaml [hunk 1]
```diff
@@ -13,4 +13,6 @@ test_1:
     sessions_per_node: 20  # Number of sessions per node.
     background_delete: False  # Enable/Disable background delete(Cortx specific).
     tool: s3bench
+    s3max_retries: 5 # Maximum number of times that a request will be retried for failures.
+    connection_timeout: 5 # Specifies a time limit in Minutes for requests made by Client.
```
Comment (priyankaborawake) [inline workload/s3/mix_io/type4_mix_object_crud_operations.yaml]: Please make it 3 mins for happy path.

Comment (priyankaborawake) [reply workload/s3/mix_io/type4_mix_object_crud_operations.yaml]: Can we have default values and input can be taken as part of test on requirement basis.(for example in degraded test.)

File: workload/s3/mix_io/type3_mix_crud_object_operations.yaml [hunk 2]
```diff
@@ -12,5 +12,7 @@ test_1:
     min_runtime: 30d  # Minimum execution durations to mark workload pass.
     sessions_per_node: 20  # Number of sessions per node.
     background_delete: False  # Enable/Disable background delete(Cortx specific).
+    s3max_retries: 5 # Maximum number of times that a request will be retried for failures.
```
Comment (priyankaborawake) [inline workload/s3/mix_io/type3_mix_crud_object_operations.yaml]: As mentioned below, we should not be taking this input for each test.

File: src/libs/tools/s3bench.py [hunk 3]
```diff
@@ -190,7 +196,12 @@ def execute_s3bench_workload(self) -> Tuple[bool, Any]:
             if self.skip_cleanup:
                 cmd += "-skipCleanup "
             if self.validate:
-                cmd += "-validate"
+                cmd += "-validate "
+            if self.s3max_retries:
+                cmd = cmd + f"-s3MaxRetries={self.s3max_retries} "
+            if self.connection_timeout:
```
Comment (priyankaborawake) [inline src/libs/tools/s3bench.py]: better to mention it as httpClientTimeout  as there are many type of timeouts supported.

File: src/libs/tools/s3bench.py [hunk 4]
```diff
@@ -190,7 +196,12 @@ def execute_s3bench_workload(self) -> Tuple[bool, Any]:
...
+                self.connection_timeout *= 60000  # conversion minutes into milliseconds.
+                cmd = cmd + f"-httpClientTimeout={self.connection_timeout}"
```
Comment (priyankaborawake) [inline src/libs/tools/s3bench.py]: single space can be added at the end.

File: src/libs/tools/s3bench.py [hunk 5]
```diff
@@ -65,6 +67,8 @@ def __init__(self, access: str, secret: str, test_id: str,
         :param skip_write: Skip write operation
         :param skip_cleanup: Skip cleanup operation
         :param validate: Validate downloaded objects
+        :param s3max_retries: Maximum number of times that a request will be retried for failures
+        :param connection_timeout: Time limit in Minutes for requests made by this Client.
```
Comment (priyankaborawake) [inline src/libs/tools/s3bench.py]: Please replace with similar to httpClientTimeout 
maybe http_client_timeout

File: scripts/s3/mixs3io/mix_object_operations.py [hunk 6]
```diff
@@ -231,6 +233,11 @@ def execute_validate_run(self, cmd: str) -> None:
 
         :param cmd: s3bench command to be executed with some workload.
         """
+        if self.s3max_retries:
```
Comment (priyankaborawake) [inline scripts/s3/mixs3io/mix_object_operations.py:239]: Please add s3bench lib and these parameters can be handled in lib.

**Review by priyankaborawake — CHANGES_REQUESTED** [review]
Hi @DhobaleVishal,
Requested few changes.

File: scripts/s3/mixs3io/mix_object_operations.py (hunk 6 above)
Comment (DhobaleVishal) [reply scripts/s3/mixs3io/mix_object_operations.py:239]: This will be taken care by @VikasKumarC while implementing adapter for all tools.

**Review by rahul-telawade — APPROVED** [review]

**Review by priyankaborawake — APPROVED** [review]

**Pull request merged**